snapcraft/storeapi/__init__.py

```diff
@@ -364,6 +364,18 @@ def get_assertion(self, snap_id, endpoint):
     def sign_developer_agreement(self, latest_tos_accepted=False):
         return self.sca.sign_developer_agreement(latest_tos_accepted)
 
+    def update_metadata(self, snap_name, metadata, arch, force):
```

Comment:
What's the purpose of `arch` here? It only seems to be used for the error.

Reply:
Exactly! The idea is to maintain the detail level of the rest of SnapNotFoundError messages.

Reply:
I see. It just strikes me as odd that the method would use a value in the error that isn't needed to do the actual work. As in, the error is providing more detail than is relevant to the failure. I might be missing something, though.